homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -117,6 +117,9 @@ def _async_migrate_options_from_data_if_missing(
         options = dict(DEFAULT_OPTIONS, **options)
         options[CONF_READ_ONLY] = data.pop(CONF_READ_ONLY, False)
 
+        # Remove CONF_USE_LOCATION as we have to use it by default
+        options.pop(CONF_USE_LOCATION, None)
```

Comment:
We shouldn't modify the stored config entry data since we want to be backwards compatible.